src/modules/clustering/hierarchical_clustering_method.cpp

```diff
@@ -0,0 +1,169 @@
+#include "modules/clustering/hierarchical_clustering_method.hpp"
+
+#include <limits>                                                 // for infinity
+
+#include "fastcluster.h"                                          // for hclust_fast
+
+
+std::vector<std::vector<Junction>> partition_junctions(std::vector<Junction> const & junctions)
+{
+    // Partition based on mate 1
+    std::vector<Junction> current_partition{};
+    std::vector<std::vector<Junction>> current_partition_splitted;
+    std::vector<std::vector<Junction>> final_partitions{};
+
+    for (Junction junction : junctions)
+    {
+        if (current_partition.empty())
+        {
+            current_partition.push_back(junction);
```

Comment:
For maximum speed:
```suggestion
            current_partition.emplace_back(std::move(junction));
```

Reply:
Good point! Just to make sure I understood correctly: In each iteration of the for-loop, a new copy of the junction is created. If we simply `push_back()`, a second (unnecessary) copy is created.
Your suggestion now moves the first junction copy directly into the vector. Another similar option would be:
`current_partition.push_back(std::move(junction));`
This should also move the junction. Is that all correct or am I missing something?

Reply:
Another option that just came to my mind.. Would that have the same effect (only one copy per junction)?
```
for (Junction & junction : junctions)
{
    ... 
    current_partition.push_back(junction);
    ...
}
```

Reply:
Yes. This is even better! One small thing, the const goes to the right: `for (Junction const & junction : junctions)`